src/thunderbots/software/ai/navigator/obstacle/obstacle.h

```diff
@@ -14,8 +16,35 @@
 class Obstacle
 {
    public:
+    /**
+     * Create an obstacle from the given polygon
+     *
+     * @param polygon
+     */
+    Obstacle(Polygon polygon);
+
     static Obstacle createRobotObstacle(const Robot& robot, bool enable_velocity_cushion);
 
+
+    /*
+     * Gets the boundary polygon around the given primitive that other robots
+     * should not enter with a buffer scaled by width_scaling and length_scaling
+     *
+     * length is along the length of the primitive and the width is perpendicular to that
+     *
+     * @param start             start position
+     * @param destination       destination position
+     * @param initial_speed     initial speed of the robot
+     * @param width_scaling     multiplicatively scales the width of obstacle
+     * @param length_scaling    multiplicatively scales the length
+     *
+     * @return a rectangular Polygon to represent the boundary around the obstacle
+     */
+    static Obstacle createVelocityObstacleWithScalingParams(Point start, Point end,
```

Comment:
IMO this function name is too verbose, we can trim it down to something like `createRobotVelocityObstacle`.